### ISSUE TITLE
Element button is missing one or more of the following attributes: aria-checked, role...

### ISSUE THREAD

**Issue by Mogget24** [validator/validator#769]
Hi,
I have this HTML checked by the validator:
`<button type="button" class="show-password js-toggle-password" aria-selected="true">Show password</button>`

The error message I got is the same of the title of this issue

A <button> doesn't need a role because it has the implicit role="button"

Thanks

**Comment by sideshowbarker** [comment]
If you have `aria-selected` then you also need to have an explicit role — and the role must be one of gridcell, option, row, tab, columnheader, rowheader, or treeitem. See the spec:

https://w3c.github.io/aria/#aria-selected

(`aria-selected` isn’t allowed with the `button` role at all.)

**Comment by Mogget24** [comment]
Thanks
